### ISSUE TITLE
gcc-4.8.5 error: declaration of ‘struct Eigen::internal::traits<double>’  template<typename T> struct traits

### ISSUE THREAD

**Issue by marlinilram** [CGAL/cgal#3557]
_Please use the following template to help us solving your issue._

## Issue Details

Many eigen error **declaration of ‘struct Eigen::internal::traits<double>’  template<typename T> struct traits** after include cgal headers. Eigen works fine with libigl before.  And it also works fine in MacOS with LLVM-10.0 and Ubuntu 18.04 with GCC-7.3.0 but fails in Ubuntu 16.04 with GCC-5.4.0 and CentOS 7 with GCC-4.8.2

**configured in header only mode with cmake**

set(CGAL_DIR ${CMAKE_CURRENT_LIST_DIR}/../3rdparty/cgal)
find_package(CGAL REQUIRED)
include(${CGAL_USE_FILE})

## Source Code

basically the same code from Point_set_shape_detection_3/efficient_RANSAC_point_access.cpp

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Ubuntu 16.04 64bit and CentOS 7 64bit
* Compiler: GCC-4.8.2 and GCC-5.4.0
* Release or debug mode: release
* Specific flags used (if any):
* CGAL version: 4.13
* Boost version: 1.53 and 1.58
* Other libraries versions if used (Eigen, TBB, etc.):

**Comment by marlinilram** [comment]
problem solved when I directly include eigen and libigl header files in my CMakeLists instead of the .cmake file shipped within libigl
